common/script/libs/shops.js

```diff
@@ -0,0 +1,241 @@
+    notes: i18n.t('dropsExplanation', language),
+  };
+  foodCategory.items = _(content.food)
+    .values()
+    .filter(food => food.canDrop || food.key === 'Saddle')
+    .map(foodItem => {
+      return {
+        key: foodItem.key,
+        text: foodItem.text(language),
+        notes: foodItem.notes(language),
+        class: `Pet_Food_${foodItem.key}`,
+        value: foodItem.value,
+        locked: false,
+        currency: 'gems',
+        purchaseType: 'food',
+      };
+    }).sortBy('key').value();
+  categories.push(foodCategory);
+
+  return categories;
+};
+
+shops.getQuestShopCategories = function getQuestShopCategories (user, language) {
+  let categories = [];
+
+  _.each(content.userCanOwnQuestCategories, type => {
+    let category = {
+      identifier: type,
+      text: i18n.t(`${type}Quests`, language),
+    };
+
+    category.items = _(content.questsByLevel)
+      .filter(quest => quest.canBuy(user) && quest.category === type)
+      .map(quest => {
+        let locked = lockQuest(quest, user);
+        return {
+          key: quest.key,
+          text: quest.text(language),
+          notes: quest.notes(language),
+          value: quest.goldValue ? quest.goldValue : quest.value,
+          currency: quest.goldValue ? 'gold' : 'gems',
+          locked,
+          unlockCondition: quest.unlockCondition,
+          drop: quest.drop,
+          boss: quest.boss,
+          collect: quest.collect,
+          lvl: quest.lvl,
+          class: locked ? `inventory_quest_scroll_${quest.key}_locked` : `inventory_quest_scroll_${quest.key}`,
+          purchaseType: 'quests',
+        };
+      }).value();
+
+    categories.push(category);
+  });
+
+  return categories;
+};
+
+shops.getTimeTravelersCategories = function getTimeTravelersCategories (user, language) {
+  let categories = [];
+  let stable = {pets: 'Pet-', mounts: 'Mount_Head_'};
+  for (let type in stable) {
+    if (stable.hasOwnProperty(type)) {
+      let category = {
+        identifier: type,
+        text: i18n.t(type, language),
+        items: [],
+      };
+
+      for (let key in content.timeTravelStable[type]) {
+        if (content.timeTravelStable[type].hasOwnProperty(key)) {
+          if (!user.items[type][key]) {
+            let item = {
+              key,
+              text: content.timeTravelStable[type][key](language),
+              class: stable[type] + key,
+              type,
+              purchaseType: type,
+              value: 1,
+              notes: '',
+              locked: false,
+              currency: 'hourglasses',
+            };
+            category.items.push(item);
+          }
+        }
+      }
+      if (category.items.length > 0) {
+        categories.push(category);
+      }
+    }
+  }
+
+  let sets = content.timeTravelerStore(user.items.gear.owned);
+  for (let setKey in  sets) {
+    if (sets.hasOwnProperty(setKey)) {
+      let set = sets[setKey];
+      let category = {
+        identifier: set.key,
```

Comment:
I'm not sure about this API. Do we really want to return each set as a separate category?

I mean, it makes a kind of sense, because the thing that is being sold is a set of items... but it still feels weird to me.